### PR TITLE
Update collections.textile

### PR DESCRIPTION
**Pull request by m4jing**
1. added `Array`
2. updated code samples according to newest English version
3. translation corrected and optimized

### CODE REVIEW

File: web/zh_cn/collections.textile [hunk 1]
```diff
@@ -31,27 +32,44 @@ h1. 基本数据结构
...
+
+h2(#Arrays). 数组 Array
+
+数组是有序的，可以包含重复项，并且可变。
+
+<pre>
+scala> val numbers = Array(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
+numbers: Array[Int] = Array(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
+
+scala> numbers(3) = 10
```
Comment (benpence) [inline web/zh_cn/collections.textile]: Can you show that the `Array` has changed?

Comment (m4jing) [inline web/zh_cn/collections.textile]: `Array` in English version is OK.
The problem is that there's no `Array` part in the current Chinese version.

So I added it and offered my translation, as well as the original code example. 😄

Comment (benpence) [inline web/zh_cn/collections.textile]: Sorry, I probably didn't communicate this well. Can you add a REPL step that evaluates the modified array like

```
scala> val numbers = Array(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)
numbers: Array[Int] = Array(1, 2, 3, 4, 5, 1, 2, 3, 4, 5)

scala> numbers(3) = 10
scala> numbers
numbers: Array[Int] = Array(1, 2, 3, 10, 5, 1, 2, 3, 4, 5)
```

Comment (m4jing) [inline web/zh_cn/collections.textile]: Of course. I'll do this soon. Thanks.

**Comment by m4jing** [comment]
@benpence I've added the REPL step. Would you please check again? Thanks a lot. 😄

**Pull request merged**